ext/spl/spl_dllist.c

```diff
@@ -722,75 +722,70 @@ PHP_METHOD(SplDoublyLinkedList, getIteratorMode)
 /* {{{ Returns whether the requested $index exists. */
 PHP_METHOD(SplDoublyLinkedList, offsetExists)
 {
-	zval              *zindex;
 	spl_dllist_object *intern;
-	zend_long               index;
+	zend_long index;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "z", &zindex) == FAILURE) {
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "l", &index) == FAILURE) {
 		RETURN_THROWS();
 	}
 
 	intern = Z_SPLDLLIST_P(ZEND_THIS);
-	index  = spl_offset_convert_to_long(zindex);
 
 	RETURN_BOOL(index >= 0 && index < intern->llist->count);
 } /* }}} */
 
 /* {{{ Returns the value at the specified $index. */
 PHP_METHOD(SplDoublyLinkedList, offsetGet)
 {
-	zval                  *zindex;
 	zend_long                   index;
 	spl_dllist_object     *intern;
 	spl_ptr_llist_element *element;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "z", &zindex) == FAILURE) {
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "l", &index) == FAILURE) {
 		RETURN_THROWS();
 	}
 
 	intern = Z_SPLDLLIST_P(ZEND_THIS);
-	index  = spl_offset_convert_to_long(zindex);
 
 	if (index < 0 || index >= intern->llist->count) {
-		zend_throw_exception(spl_ce_OutOfRangeException, "Offset invalid or out of range", 0);
+		zend_argument_error(spl_ce_OutOfRangeException, 1, "is out of range");
 		RETURN_THROWS();
 	}
 
 	element = spl_ptr_llist_offset(intern->llist, index, intern->flags & SPL_DLLIST_IT_LIFO);
-
-	if (element != NULL) {
-		zval *value = &element->data;
-
-		ZVAL_COPY_DEREF(return_value, value);
-	} else {
-		zend_throw_exception(spl_ce_OutOfRangeException, "Offset invalid", 0);
+	if (element == NULL) {
+		zend_argument_error(spl_ce_OutOfRangeException, 1, "is an invalid offset");
+		RETURN_THROWS();
 	}
+
+	zval *value = &element->data;
+	ZVAL_COPY_DEREF(return_value, value);
 } /* }}} */
 
 /* {{{ Sets the value at the specified $index to $newval. */
 PHP_METHOD(SplDoublyLinkedList, offsetSet)
 {
-	zval                  *zindex, *value;
-	spl_dllist_object     *intern;
+	zend_long index;
+	zend_bool index_is_null = 1;
+	zval *value;
+	spl_dllist_object *intern;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "zz", &zindex, &value) == FAILURE) {
-		RETURN_THROWS();
-	}
+	ZEND_PARSE_PARAMETERS_START(2, 2)
+		Z_PARAM_LONG_OR_NULL(index, index_is_null)
+		Z_PARAM_ZVAL(value)
+	ZEND_PARSE_PARAMETERS_END();
```

Comment:
`"l!z"` would also work, no?

Reply:
Yep.. I've got no idea what happened to me that I forgot it.